src/parser.cpp

```diff
@@ -272,13 +314,27 @@ namespace ada::parser {
             ada_log("RELATIVE_SCHEME otherwise");
             // Set url’s username to base’s username, url’s password to base’s password, url’s host to base’s host,
             // url’s port to base’s port, url’s path to a clone of base’s path, and url’s query to base’s query.
-            url.username = base_url->username;
-            url.password = base_url->password;
-            url.host = base_url->host;
-            url.port = base_url->port;
-            url.path = base_url->path;
+            if constexpr (result_type_is_ada_url) {
+              url.username = base_url->username;
+              url.password = base_url->password;
+              url.host = base_url->host;
+              url.port = base_url->port;
+              url.path = base_url->path;
+              url.query = base_url->query;
+            } else  {
+              // TODO: This is likely not correct. There is a difference between having an empty username
```

Comment:
@anonrig  We do update_base_username even if there is no username in the base. Likely a bug.